### ISSUE TITLE
How do you manage which files are tracked and which not?

### ISSUE THREAD

**Issue by psiho** [TheLocehiliosan/yadm#244]
I'm using yadm for a few weeks and quite happy with it. But one small piece of my workflow doesn't feel right. Maybe you can point me to the right direction.

My home folder is filling up with files and folders to track, and to ignore. At the start, I knew everything by heart but after settling down and a week passed, I don't really remember which file is tracked and which not, and if I installed something new this week that should be tracked.

So what I do not is use `yadm list` and check if I missed something. However, if I do it in the root of my home folder, it's unreadable. Too much of the stuff. If I go folder by folder and compare to catch changes I forgot to `yadm add`... it's a slow and tedious process.

What's your workflow? Trouble is, all files are gitignored by default. Then you add some. But if system adds some more they are gitignored just like the ones I personally ignored. In ideal world, I could use 3 statuses: `added`, `ignored` and `untracked`. Then I would periodically just go through "untracked" list.

**Comment by TheLocehiliosan** [comment]
Does this help?

# How can I display untracked files with a yadm status command?

By default, yadm is configured to ignore untracked files when displaying a status. You can use the `-u` paramter on the status command to alter how untracked files are treated. `-unormal` will show untracked files and directories.

```
yadm status -unormal
```

If you want to change the default treatment, you can change this configuration on the yadm repository itself.

``` 
yadm gitconfig --unset status.showUntrackedFiles
```

**Comment by psiho** [comment]
Uh, very slow answer of mine, but back on this....

I was using -u/-untracked before, but again, it gives me a huge 3 screen list. Point is, it's very time consuming to go through it every week or so while building the system, and very error-prone later because it's easy to miss a file that needs to be tracked.

In example, my ~/temp/  folder is untracked and it should be. But it also appears in the list every time I do yadm status -unormal. Problem is, it appears together with a hundred other items that may or may not be untracked for a good reason. In example, I used Heroku for the first time a few weeks ago, so '.local/share/heroku' appears on this list too. And I do want to track this folder with 'yadm add'. And it's not that easy to spot it, it can be easily missed.

I can't imagin this is a regular workflow. I must be missing something.

**Comment by erijo** [comment]
Have you tried adding the directories you want to ignore to `~/.gitignore`. To ignore ~/temp/ you could add `/temp/` to that file.

**Comment by erijo** [comment]
Does the use of .gitignore + -u solve this issue @psiho?

**Comment by tmewett** [comment]
The fundamental problem here is that in Git, and hence yadm, there is no such concept as "tracking a directory." You add individual files and Git creates and deletes directories as needed. This means there is no workflow in Git to create a file and have it auto-added because it's inside a certain directory.

I wonder if it's feasible for yadm to handle this somehow. I often want to sync entire directories, such as my notes or my Fish functions. It's a burden to have to remember to add them when in the manual-symlinking, no-yadm workflow I can just symlink the whole directory.

I suppose one could use a yadm pre-commit hook which adds all your "whole directories". You would have to remember to add a dir to the hook when you create one, but you don't have to remember for files inside them.

**Comment by psiho** [comment]
@tmewett yes, I 2nd that. I mean, it is completely outside of the scope of yadm I guess, but it can serve as an idea for a product. Yadm is around Git, which is a good thing, and why I installed it in the first place. But in practice, at least for me, I ended up using just Git with some tricks because this was not manageable, at least not more than regular Git. With Git, at least I don't need to remember some more commands and settings, it's a standard tool I use every day. And config management I do rare enough to keep forgetting small nuances like this. It's too much work (mentally) to grasp what went to Git and what needs to be added, since months ago.

I tried more than a few, but at the end I don't think I'll be using any solution that does not (almost) fully automate this. Point the tool to some paths (like '~/.config') and then on each run it reports what changed, what is added there and if I want to include it or exclude it (permanently) without asking again, push it, pull it. Add some management of that list for changes (like when I  uninstalled something that managed before) and basically thats it.
